compatibility/sandbox-migration/runner/Migration/Types.hs

```diff
@@ -42,4 +46,33 @@ data Tuple2 a b = Tuple2
   , _2 :: b
   } deriving (Eq, Generic, Show)
 
-instance (A.FromJSON a, A.FromJSON b) => A.FromJSON (Tuple2 a b)
\ No newline at end of file
+instance (A.FromJSON a, A.FromJSON b) => A.FromJSON (Tuple2 a b)
+
+data TemplateId = TemplateId
+  { moduleName :: !String
+  , entityName :: !String
+  } deriving (Eq, Show)
+
+data Event
+  = Created !ContractId !TemplateId !A.Value
```

Comment:
What does the `!` stand for?

Reply:
It makes the field strict rather than lazy which is the default. Shouldn’t matter much here but I consider it good practice to mark fields strict when it makes sense.

Reply:
Got it, thanks.